### PR TITLE
avoiding duplicate documents

### LINKED ISSUE #203: Duplicate names of projects and components can be added
Maybe this will create confusion if several users would be able to add projects or components with the same name?

### PR DESCRIPTION
**Pull request by heydenreich**
check in backend and display error in frontend in the cases when a user tries to:
* add a project with name and version equal to an existing project
* add a component with name equal to an existing component
* add a release with name and version equal to an existing release
* closes #203

### CODE REVIEW

File: backend/src-common/src/main/java/com/siemens/sw360/datahandler/db/ReleaseRepository.java [hunk 1]
```diff
@@ -99,6 +99,14 @@ public ReleaseRepository(DatabaseConnector db, VendorRepository vendorRepository
         return makeSummary(SummaryType.SHORT, queryForIdsByPrefix("byname", name));
     }
 
+    public List<Release> searchByNameAndVersion(String name, String version){
+        List<Release> releasesMatchingName = queryView("byname", name);
```
Comment (alex-evo) [inline backend/src-common/src/main/java/com/siemens/sw360/datahandler/db/ReleaseRepository.java]: why use `queryView()` and then `makeSummary...` when you could `searchByName` and filter that by version?

File: backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftExchange.java [hunk 2]
```diff
@@ -204,7 +204,7 @@ public String addVendor(Vendor vendor) {
     public String addComponent(Component component, User user) {
         String componentId = null;
         try {
-            componentId = getComponentClient().addComponent(component, user);
+            componentId = getComponentClient().addComponent(component, user).getId();
```
Comment (alex-evo) [inline backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftExchange.java]: You're sure that there will be duplicate error when importing from bdp, right?

File: frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/components/ComponentPortlet.java [hunk 3]
```diff
@@ -796,17 +804,25 @@ public void updateRelease(ActionRequest request, ActionResponse response) throws
...
+                    switch(status){
+                        case SUCCESS:
+                            response.setRenderParameter(RELEASE_ID, summary.getId());
+                            String successMsg = "Release " + printName(release) + " added successfully";
+                            SessionMessages.add(request, "request_processed", successMsg);
+                            response.setRenderParameter(PAGENAME, PAGENAME_EDIT_RELEASE);
+                            break;
+                        case DUPLICATE:
+                            setSW360SessionError(request, ErrorMessages.RELEASE_DUPLICATE);
+                            response.setRenderParameter(PAGENAME, PAGENAME_DETAIL);
```
Comment (alex-evo) [inline frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/components/ComponentPortlet.java]: So if a duplicate exists for whatever reason (mistyped version, for example), the user gets thrown back to the component display and loses any data entered in the release edit form. Same effect with projects and components. In contrast, license edit form retains entered information when there's a duplicate and lets the user correct his input.

File: backend/src-common/src/main/java/com/siemens/sw360/datahandler/db/ReleaseRepository.java (hunk 1 above)
Comment (heydenreich) [inline backend/src-common/src/main/java/com/siemens/sw360/datahandler/db/ReleaseRepository.java]: I'll change it.

**Comment by mcjaeger** [comment]
@heydenreich please assign me, if you reviewed alex review

**Comment by heydenreich** [comment]
@mcjaeger : I'm still busy with it. I found a NPE with cause somewhere in the bdp-import and a thrift-error message that had nothing to do with the actual cause... 
Now, I'm busy with the frontend - it will take some more time.

**Comment by mcjaeger** [comment]
do not worry, plenty to do for me as well, but thanks for letting me know. Just indicating that I am willing to take it over then.

**Comment by alex-evo** [comment]
Test failure:

1. duplicate a release from the component's release view
1. change Programming Languages and/or Operating Systems and/or Vendor (maybe other fields, too, but not name or version)
1. Add Release

The application displays the duplicate error and stays on the edit page, but the values entered in Programming Languages etc. are lost.

**Pull request merged**